components/Recipients.tsx

```diff
@@ -17,6 +18,22 @@ export default function Recipients({ yearlyYield }: Props) {
   const { openModal } = useContext(ModalContext);
   const [selected, setSelected] = useState({} as { [key: string]: boolean });
   const [search, setSearch] = useState("");
+  const [fundingChoices, setFundingChoices] = useState<{
+    [key: string]: number;
+  }>({});
+  const [totalYield, setTotalYield] = useState(0);
+
+  // Fetch and set funding choices on component mount
+  useEffect(() => {
```

Comment:
Perhaps a good opportunity to useSWR (https://swr.vercel.app/) - optional comment.